doc/concepts.md

```diff
@@ -1,190 +1,267 @@
+
+Figure 2 shows the datamodel hierarchy implemented in DLite.
+As a user, you will almost always deal with only entities or data
+instances, but the hierarchy gives DLite a strong and theoretically
+well-defined schema.
+
+The actual data or *Data instances* are instances of the *Entity* that
+describes them.
+*Entities* are instances of the *EntitySchema* The *EntitySchema* is
+an instance of the *BasicMetadataSchema*, is an instance of itself
+(meaning that it can be used to describe itself).
+Hence, in DLite is **everything is an instance**.
+This has a practical implication that the API for instances can be applied
+to all metadata as well.
+Since the BasicMetadataSchema can describe itself, no more abstraction
+levels are needed, making the DLite metadata schema complete and well-defined.
+
+From Figure 2, one can also see that *[collections]* are simply
+instances of the *CollectionEntity*.
+
+In DLite all instances that describe other instances (of a lower abstraction
+level) are called *metadata*.
+
+Compared to ontologies, *data instances* correspond to OWL individuals
+while *entities* correspond to OWL classes.
+OWL, which is based on first order logic, do not have concepts corresponding
+to the higher abstraction levels of EntitySchema and BasicMetadataSchema.
+
+A (so far unexplored possibility) with such a datamodel hierarchy is
+that it can enable cross-platform semantic interoperability between
+independent systems that describe their datamodel hierarchies using a
+common BasicMetadataSchema.
+Of course, this requires a common agreement on the BasicMetadataSchema.
+
+
+### Entities
+An entity can be a single thing or object that represents something
+physical or nonphysical, concretely or abstract.
+The entity contains information about the data that constitutes the
+state of thing it describes.
+The entity does not contain the actual data, but describes what the
+different data fields are, in terms of name, data types, units,
+dimensionality etc.
+Information about data is often called metadata.
+Formal meta data enables for the correct interpretation of a set of
+data, which otherwise would be unreadable.
+
+An example of an entity is 'Atom', which can be defined as something
+that has a position, an atomic number (which characterizes the
+chemical element), mass, charge, etc.
+Another example of a completely different kind of entity can be a data
+reference-entity with properties such as name, description, license,
+access-url, media-type, format, etc). The first entity is suitable as
+an object in a simulation code, while the latter is more suitable for
+a data catalog distribution description (e.g. [dcat:Distribution]).
+Entities allows for describing many aspects of the domain.
+While each entity describes a single unit of information, a set of
+entities can describe the complete domain.
+
+
+#### Uniqueness & immutability
+To ensure consistency, no entity (or other metadata) should never be
```

Comment:
either: "an entity ... should never be changed..." or "no entity .... should ever be changed..."